test/e2e/app-dir/ppr-metadata-streaming/app/dynamic-page/page.tsx

```diff
@@ -0,0 +1,31 @@
+import { cookies } from 'next/headers'
+import { Suspense } from 'react'
+import Link from 'next/link'
+
+// Dynamic usage in page, wrapped with Suspense boundary
+export default function Page() {
+  return (
+    <div>
+      <h1>Dynamic Page</h1>
+      <Suspense fallback={<div>Loading...</div>}>
+        <SubComponent />
+      </Suspense>
+      <Link href="/">Home</Link>
+    </div>
+  )
+}
+
+async function SubComponent() {
+  const cookieStore = await cookies()
+  await new Promise((resolve) => setTimeout(resolve, 500))
+  const cookie = await cookieStore.get('test')
+  return <div>Cookie: {cookie?.value}</div>
+}
+
+export async function generateMetadata() {
+  // Slow but static metadata
```

Comment:
I think we need a case that generates a partial shell with dynamic metadata. For example, if `app/dynamic-page/layout.tsx` had some content and then wrapped `children` in `Suspense`. From what I can tell, when you do that we aren't serving a response that has the `title` in `head` and instead it's waiting for hydration to do the insertion.